Correlation api

Alexanderlacuna opened the pull request:
Correlation Endpoints for genenetwork3

Alexanderlacuna committed:
add file for correlation api

Alexanderlacuna committed:
register initial correlation api

Alexanderlacuna committed:
add correlation package

Alexanderlacuna committed:
add function  for getting page data

Alexanderlacuna committed:
delete loading page api

Alexanderlacuna committed:
modify code for correlation

Alexanderlacuna committed:
add tests folder for correlations

robwwilliams commented:
Fast progress Alex!

Alexanderlacuna committed:
fix error in correlation api

Alexanderlacuna committed:
add tests for correlation

Alexanderlacuna committed:
add tests for  correlation loading data

Alexanderlacuna committed:
add module for correlation computations

Alexanderlacuna committed:
modify api to return json when computing correlation

Alexanderlacuna committed:
add tests for computing correlation

Alexanderlacuna committed:
modify code for loading correlation data

Alexanderlacuna committed:
modify tests for correlation computation

Alexanderlacuna committed:
test loading correlation data using api endpoint

Alexanderlacuna committed:
add tests for asserting error in creating Correlation object

Alexanderlacuna committed:
add do correlation method

Alexanderlacuna committed:
add dummy tests for do_correlation method

Alexanderlacuna committed:
delete unused modules

Alexanderlacuna committed:
add tests for creating trait and dataset

Alexanderlacuna committed:
add intergration test for correlation api

Alexanderlacuna committed:
add tests for correlation api

Alexanderlacuna committed:
edit docorrelation method

Alexanderlacuna committed:
modify integration tests for correlation api

Alexanderlacuna committed:
modify tests for show_corr_results

Alexanderlacuna committed:
add create dataset function

Alexanderlacuna committed:
pep8 formatting and fix return value for api

Alexanderlacuna committed:
add more test data for doing correlation

Alexanderlacuna committed:
modify tests for correlation

Alexanderlacuna committed:
pep8 formatting

Alexanderlacuna committed:
add getting formatted corr type method

Alexanderlacuna committed:
import json library

add process samples method for correlation

Alexanderlacuna committed:
fix issue with sample_vals key_error

Alexanderlacuna committed:
create utility module for correlation

Alexanderlacuna committed:
refactor endpoint for /corr_compute

Alexanderlacuna committed:
add test and mocks for compute_correlation function

Alexanderlacuna committed:
add compute correlation function  and pep8 formatting

Alexanderlacuna committed:
move get genofile samplelist to utility module

Alexanderlacuna committed:
refactor code for CorrelationResults object

Alexanderlacuna committed:
pep8 formatting for module

Alexanderlacuna committed:
remove CorrelationResults from Api

BonfaceKilz commented on gn3/app.py:
```diff
@@ -27,4 +28,5 @@ def create_app(config: Union[Dict, str, None] = None) -> Flask:
             app.config.from_pyfile(config)
     app.register_blueprint(general, url_prefix="/")
     app.register_blueprint(gemma, url_prefix="/gemma")
+    app.register_blueprint(correlation)
```
**issue:** You should probably use a url prefix here. So that a user hits: _localhost:8080/correlation/_

BonfaceKilz reviewed (commented):
Nice work! Once this is done, we can clean these abstractions up!

Alexanderlacuna committed:
add base package

initialize data_set module with create_dataset,redis and Dataset_Getter

Alexanderlacuna committed:
set dataset_structure if redis is empty

Alexanderlacuna committed:
add callable for DatsetType

Alexanderlacuna committed:
add set_dataset_key method If name is not in the object's dataset dictionary

Alexanderlacuna committed:
add Dataset object and MrnaAssayDataSet

Alexanderlacuna committed:
add db_tools

Alexanderlacuna committed:
add mysql client

Alexanderlacuna committed:
add DatasetGroup object

Alexanderlacuna committed:
add species module

Alexanderlacuna committed:
get mapping method

Alexanderlacuna committed:
import helper functions and new dataset

Alexanderlacuna committed:
add connection to db before request

Alexanderlacuna committed:
add helper functions

Alexanderlacuna committed:
add logger module

Alexanderlacuna committed:
add get_group_samplelists module

Alexanderlacuna committed:
add logger for debug

Alexanderlacuna committed:
add code for adding sample_data

Alexanderlacuna committed:
pep8 formatting

Alexanderlacuna committed:
Add chunks module

Alexanderlacuna committed:
add correlation helper module

Alexanderlacuna committed:
add  get_sample_r_and_p_values method

add get_header_fields function

Alexanderlacuna committed:
add generate corr json method

Alexanderlacuna committed:
add function to retrieve_trait_info

Alexanderlacuna committed:
remove comments and clean up code in show_corr_results

Alexanderlacuna committed:
remove comments and clean up code for data_set module

Alexanderlacuna committed:
pep8 formatting for helper_functions module

Alexanderlacuna committed:
pep8 formatting for trait module

Alexanderlacuna committed:
add module for species

Alexanderlacuna committed:
add Temp Dataset Object

Alexanderlacuna committed:
add Phenotype Dataset

Alexanderlacuna committed:
add Genotype Dataset

Alexanderlacuna committed:
add rettrieve sample_sample_data method

Alexanderlacuna committed:
add webqtlUtil module

Alexanderlacuna committed:
add do lit correlation for all traits

Alexanderlacuna committed:
add webqtlCaseData:Settings not ported

Alexanderlacuna committed:
return the_trait for create trait method

Alexanderlacuna committed:
add correlation_test json data

Alexanderlacuna committed:
add tests fore show corr results

Alexanderlacuna committed:
add dictfier package

Alexanderlacuna committed:
add tests for show_corr_results

Alexanderlacuna committed:
add assertion for trait_id

Alexanderlacuna committed:
refactor code for show_corr_results

Alexanderlacuna committed:
add test file for compute_corr intergration tests

Alexanderlacuna committed:
add scipy dependency

Alexanderlacuna committed:
refactor show_corr_results object

add do lit correlation for trait_list

Alexanderlacuna committed:
add hmac module

Alexanderlacuna committed:
add bunch module:Dictionary using object notation

Alexanderlacuna committed:
add correlation functions

Alexanderlacuna committed:
add rpy2 dependency

Alexanderlacuna committed:
add hmac module

Alexanderlacuna committed:
add MrnaAssayTissueData object and get_symbol_values_pairs function

Alexanderlacuna committed:
add config module

Alexanderlacuna committed:
add get json_results method

Alexanderlacuna committed:
pep8 formatting remove comments

Alexanderlacuna committed:
add config file

Alexanderlacuna committed:
add db package

Alexanderlacuna committed:
refactor correlatio compuatation module

Alexanderlacuna committed:
add do tissue correlation for trait list

Alexanderlacuna committed:
add  do lit correlation for all traits

Alexanderlacuna committed:
add do tissue correlation for all traits

Alexanderlacuna committed:
add do_bicor for bicor method

Alexanderlacuna committed:
raise error for when initital start vars is None

Alexanderlacuna committed:
add support for both form and json data when for correlation input

Alexanderlacuna committed:
remove print statement and pep8 formatting

Alexanderlacuna committed:
add default settings file

Alexanderlacuna committed:
add tools module for locate_ignore_error

Alexanderlacuna committed:
refactor code remove comments for trait module

Alexanderlacuna committed:
Add new test data for  computing correlation

Alexanderlacuna committed:
pep8 formatting and use pickle

Alexanderlacuna committed:
refactor function for filtering form/json data

Alexanderlacuna committed:
remove unused imports

Alexanderlacuna committed:
remove mock functions in correlation_utility module

Alexanderlacuna committed:
refactor tests for compute correlation and pep8 formatting

Alexanderlacuna committed:
add tests for show_correlation results

Alexanderlacuna committed:
modify tests for show_corr_results

Alexanderlacuna committed:
add json files for tests

Alexanderlacuna committed:
pep8 formatting for show_corr_results

Alexanderlacuna committed:
Todo:Lint base files

Alexanderlacuna committed:
pylint for intergration tests

Alexanderlacuna committed:
add test module for test_corr_helpers

Alexanderlacuna committed:
Add test chunk module

Alexanderlacuna committed:
lint utility package

Alexanderlacuna committed:
refactoring and pep8 formatting

Alexanderlacuna committed:
implement simple metric for correlation

Alexanderlacuna committed:
add  hmac utility file

Alexanderlacuna committed:
add correlation prefix

Alexanderlacuna committed:
fix merge conflict

Alexanderlacuna committed:
minor fixes for endpoints

Alexanderlacuna committed:
Merge branch 'main' into correlation-api

Alexanderlacuna committed:
import:python-scipy,python-sqlalchemy from guix

Alexanderlacuna committed:
add python mysqlclient

Alexanderlacuna committed:
fix merge conflict in app.py

Alexanderlacuna committed:
remove pkg-resources from requirements

Alexanderlacuna committed:
merge correlation-api remote branch

Alexanderlacuna committed:
add python-rpy3 from guix

Alexanderlacuna committed:
refactor code for species module

Alexanderlacuna committed:
pep8 formatting and refactor code

Alexanderlacuna committed:
add tests for genereating correlation results

Alexanderlacuna committed:
lint correlation functions

Alexanderlacuna committed:
fix failing tests for show_corr_results

Alexanderlacuna committed:
add new correlation test data fix errors

Alexanderlacuna committed:
fix issues related to getting group samplelists

Alexanderlacuna committed:
refactor intergration tests for correlation

Alexanderlacuna committed:
add todo  for refactoring_wanted_inputs

Alexanderlacuna committed:
replace custom Attribute setter with SimpleNamespace

Alexanderlacuna committed:
comparison of sample r correlation results btwn genenenetwork2 and genenetwork3

Alexanderlacuna committed:
delete AttributeSetter

Alexanderlacuna committed:
test request for /api/correlation/compute_correlation took 18.55710196495056 Seconds

Alexanderlacuna committed:
refactor tests and show_correlation results

Alexanderlacuna committed:
remove  unneccessary comments and print statements

Alexanderlacuna committed:
edit requirement txt file

Alexanderlacuna committed:
api/correlation took 114.29814600944519 Seconds for correlation resullts:20000

 - corr-type:lit

- corr-method:pearson

corr-dataset:corr_dataset:HC_M2_0606_P

Alexanderlacuna committed:
capture SQL_URI and GENENETWORK FILES path

Alexanderlacuna committed:
pep8 formatting edit && remove print statements

Alexanderlacuna committed:
delete filter_input function

update test and data for correlation

Alexanderlacuna committed:
add docstring for required correlation_input

Alexanderlacuna committed:
/api/correlation took 12.905632972717285 Seconds

 *  pearson

 * lit

 *dataset:HX_M2_0606_P

trait_id :1444666

p_range:(lower->-0.60,uppper->0.74)

corr_return_results: 100

Alexanderlacuna committed:
update integration and unittest for correlation

Alexanderlacuna committed:
add simple markdown docs for correlation

Alexanderlacuna committed:
update docs

Alexanderlacuna committed:
add tests and catch for invalid correlation_input

Alexanderlacuna committed:
minor fix for api

BonfaceKilz committed:
Remove jupyter from deps

BonfaceKilz committed:
guix.scm: Remove duplicate entry

BonfaceKilz committed:
guix.scm: Add extra action items as comments

BonfaceKilz committed:
Trim requirements.txt file

BonfaceKilz commented:
We'll get rid of the gn2 datastructures(i.e. dataset) borrowed from gn2 since it makes things very unmaintainable and difficult to read.

BonfaceKilz commented:
Also, we'll live with things being Red for a while some of these stuff are reworked...

robwwilliams commented:
Spring cleaning in progres.

On Sat, Mar 13, 2021 at 4:04 AM BonfaceKilz ***@***.***>
wrote:

> Also, we'll live with things being Red for a while some of these stuff are
> reworked...
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork3/pull/2#issuecomment-798073591>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC55V7FNKFNRULP3U7H6QETTDM2B3ANCNFSM4XZXU7DA>
> .
>
-- 
Rob

Robert W. Williams, Ph.D.
Chair: Department of Genetics, Genomics and Informatics
University of Tennessee Health Science Center
71 S Manassas St, Memphis TN 38103-3310,  TSRB Room 405
Tele/Cell: 901 604 4752
EMAIL: ***@***.***  or  ***@***.***

zsloan commented:
Yeah, for the purposes of the API there's no need to continue using them
(even for "regular" GN2 it probably would have been better to do that
differently).

On Sat, Mar 13, 2021 at 4:03 AM BonfaceKilz ***@***.***>
wrote:

> We'll get rid of the gn2 datastructures(i.e. dataset) borrowed from gn2
> since it makes things very unmaintainable and difficult to read.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/genenetwork/genenetwork3/pull/2#issuecomment-798072587>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQJGCAPVMNBGW6FQC2DMDTDMZ6LANCNFSM4XZXU7DA>
> .
>

BonfaceKilz commented:
zsloan ***@***.***> writes:

> Yeah, for the purposes of the API there's no need to continue using them
> (even for "regular" GN2 it probably would have been better to do that
> differently).
>
We had that conversation yesterday when I merged
the work. We'll replace most of that in a more
coherent way, something similar to what we've done
with Gemma.

> On Sat, Mar 13, 2021 at 4:03 AM BonfaceKilz ***@***.***>
> wrote:
>
>> We'll get rid of the gn2 datastructures(i.e. dataset) borrowed from gn2
>> since it makes things very unmaintainable and difficult to read.
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/genenetwork/genenetwork3/pull/2#issuecomment-798072587>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AANQJGCAPVMNBGW6FQC2DMDTDMZ6LANCNFSM4XZXU7DA>
>> .
>>

-- 
Bonface M. K. D4F09EB110177E03C28E2FE1F5BBAE1E0392253F
Humble GNU Emacs User / Bearer of scheme-y parens
Curator: <https://upbookclub.com> / Twitter: @BonfaceKilz
